### ISSUE TITLE
FDS+Evac - Last Time Step

### ISSUE THREAD

**Issue by gjaeger** [firemodels/fds#4529]
Version          : FDS 6.5.2
Revision         : Git-r21-0-g4e9103f
Revision Date    : Wed Aug 24 16:29:46 2016 -0400
Compilation Date : Wed 08/24/2016  10:28 PM

 FDS+Evac Evacuation Module
 FDS+Evac Version         : 2.5.1

Problem:
The calculation of the last time step takes a very long time (160 seconds).  The remaining calculation time is very short (10 seconds). I can reproduce the problem with different cases on different operating systems (Win7, OSX, Linux).

example inputfile:
[exercise_01_fds.txt](https://github.com/firemodels/fds/files/633344/exercise_01_fds.txt)

example cpu-outputfile:
[exercise_01_cpu.txt](https://github.com/firemodels/fds/files/633349/exercise_01_cpu.txt)

Any idea?

**Comment by tkorhon1** [comment]
I have already corrected this. It should be in the latest source code. I corrected this about one month ago.

The issue was: Kevin(?) changed the main time stepping loop in main.f90 so that the last time step was modified so that the ending time of the simulation would be exactly T_END. Previously it was anything in the range [T_END, T_END+DT), where DT was the time step. This made the internal evacuation simulation time step to be very very tiny and there is a loop over this internal evacuation time step.

So, compile the latest source code and check that everything goes nicely.

Below is my own compilation (I took the revision stuff away, because it
might be misleading, i.e., I do not compile in a git-directory, so I put
manually the revision information to the makefile and that might 
sometimes be old information). The problem does not exist in this
version anymore.

 Fire Dynamics Simulator

 Current Date     : December  8, 2016  13:13:28
 Version          : FDS 6.5.3
 Compilation Date : Nov 09, 2016  15:33:57

 MPI Enabled; Number of MPI Processes:          1
 OpenMP Enabled; Number of OpenMP Threads:   4

 MPI version: 3.0
 MPI library version: Open MPI v1.8.3, package: Open MPI tstopi@espvm5m030.ad.vt
t.fi Distribution, ident: 1.8.3, Sep 25, 2014

 Job TITLE        : Exercise No 1
 Job ID string    : exercise_01

 Time Step:    -49,    Evacuation Initialization Time:    -0.490 s
 Time Step:      1, Simulation Time:      0.05 s
 Time Step:      2, Simulation Time:      0.10 s
 Time Step:      3, Simulation Time:      0.15 s
 Time Step:      4, Simulation Time:      0.20 s
 Time Step:      5, Simulation Time:      0.25 s
 Time Step:      6, Simulation Time:      0.30 s
 Time Step:      7, Simulation Time:      0.35 s
 Time Step:      8, Simulation Time:      0.40 s
 Time Step:      9, Simulation Time:      0.45 s
 Time Step:     10, Simulation Time:      0.50 s
 Time Step:     20, Simulation Time:      1.00 s
 Time Step:     30, Simulation Time:      1.50 s
 Time Step:     40, Simulation Time:      2.00 s
 Time Step:     50, Simulation Time:      2.50 s
 Time Step:     60, Simulation Time:      3.00 s
 Time Step:     70, Simulation Time:      3.50 s
 Time Step:     80, Simulation Time:      4.00 s
 Time Step:     90, Simulation Time:      4.50 s
 Time Step:    100, Simulation Time:      5.00 s
 Time Step:    200, Simulation Time:     10.00 s
 Time Step:    300, Simulation Time:     15.00 s
 Time Step:    400, Simulation Time:     20.00 s
 Time Step:    500, Simulation Time:     25.00 s
 Time Step:    600, Simulation Time:     30.00 s
 Time Step:    700, Simulation Time:     35.00 s
 Time Step:    800, Simulation Time:     40.00 s
 Time Step:    900, Simulation Time:     45.00 s
 Time Step:   1000, Simulation Time:     50.00 s
 Time Step:   1100, Simulation Time:     55.00 s
 Time Step:   1200, Simulation Time:     60.00 s
 Time Step:   1201, Simulation Time:     60.00 s
MPI process      0 has completed

STOP: FDS completed successfully (CHID: exercise_01)

TimoK
